components/SubmissionComments.tsx

```diff
@@ -34,15 +41,83 @@ export const SubmissionComments: React.FC<{
     })
   }
 
+  const handleStartEditing = (index: number, initialInput: string) => {
+    if (editing !== null) setShow(true)
+    else {
+      setEditing(index)
+      setCurrInput(initialInput)
+    }
+  }
+
+  const handleUpdateComment = async (c: Comment, newContent: string) => {
+    const update = updateCache({
+      submissionId: submission.id,
+      content: newContent,
+      lessonId: submission.lessonId,
+      challengeId: submission.challengeId,
+      userId: submission.user.id
+    })
+
+    await editComment({
+      variables: {
+        id: c.id,
+        content: newContent
+      },
+      update
+    })
+
+    setEditing(null)
+    setCurrInput('')
+  }
+
+  const handleDiscardChange = () => {
+    setEditing(null)
+    setCurrInput('')
+  }
+
+  const handleModalClose = () => {
+    setShow(false)
+  }
+
+  const modalText =
+    'You can only edit one comment in a single comment chain at a time.'
+
+  //to make sure comments load in correct order
+  const submissionsComments = _.orderBy(comments, ['id'], ['asc'])
+
   return (
     <>
-      {comments.map((c, i) => (
+      <Modal show={show} onHide={handleModalClose}>
+        <Modal.Header>
+          <Modal.Title>Error</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>{modalText}</Modal.Body>
+        <Modal.Footer>
+          <Button
+            onClick={handleModalClose}
+            type="info"
+            size="lg"
+            color="white"
+          >
+            Close
+          </Button>
+        </Modal.Footer>
+      </Modal>
+      {submissionsComments.map((c, i) => (
         <div
           key={`${c.content}${i}`}
           className="border border-lightgray rounded my-1"
         >
           <div className="ms-3">
-            <Markdown wrapper="code_wrapper">{c.content}</Markdown>
+            {editing === i ? (
+              <MdInput
```

Comment:
Some `padding-top` + `margin-left` will make the input box centered like:

<img width="859" alt="image" src="https://user-images.githubusercontent.com/35906419/189523659-1e981171-3cb6-462e-86eb-63abfb3071e9.png">

Currently, it's so close to the top border of its container.